packages/jellyfish-api-core/src/category/poolpair.ts

```diff
@@ -153,10 +153,12 @@ export class PoolPair {
    * @param {string} metadata.to address of the owner of tokenTo
    * @param {string} metadata.tokenTo swap to token {symbol/id}
    * @param {number} [metadata.maxPrice] acceptable max price
-   * @return {Promise<string>} formatted as 'amount@token' swapped
+   * @param {'auto' | 'direct' | 'composite' | Array<string>} [path] swap path to use, defaults to auto. Provide array of poolpair IDs to set path manually.
+   * @param {boolean} [verbose] return pool path used, defaults to false
+   * @return {Promise<string | TestPoolSwapVerboseResult>} formatted as 'amount@token' swapped
    */
-  async testPoolSwap (metadata: PoolSwapMetadata): Promise<string> {
-    return await this.client.call('testpoolswap', [metadata], 'bignumber')
+  async testPoolSwap (metadata: PoolSwapMetadata, path: 'auto' | 'direct' | 'composite' | string[] = 'auto', verbose: boolean = false): Promise<string | TestPoolSwapVerboseResult> {
```

Comment:
Remember to document this change in the md file as well.

Reply:
You can also use multiple function signature to indicate what will be returned given which parameters, so that dev will have a better experience. Something like `getAccount` in https://github.com/JellyfishSDK/jellyfish/blob/main/packages/jellyfish-api-core/src/category/account.ts#L136-L173.